trackpy/feature.py

```diff
@@ -361,18 +362,19 @@ def _refine(raw_image, image, radius, coords, max_iterations,
                 upper_bound = np.array(image.shape) - 1 - radius
                 new_coord = np.clip(new_coord, radius, upper_bound).astype(int)
                 # Update slice to shifted position.
-                rect = [slice(c - rad, c + rad + 1)
+                rect = [slice(int(round(c - rad)), int(round(c + rad + 1)))
                         for c, rad in zip(new_coord, radius)]
                 neighborhood = mask*image[rect]
 
             # If we're off by less than half a pixel, interpolate.
             else:
-                # Here, coord is a float. We are off the grid.
-                neighborhood = ndimage.shift(neighborhood, -off_center,
-                                             order=2, mode='constant', cval=0)
-                new_coord = coord + off_center
-                # Disallow any whole-pixels moves on future iterations.
-                allow_moves = False
+                break
+                # # Here, coord is a float. We are off the grid.
+                # neighborhood = ndimage.shift(neighborhood, -off_center,
+                #                              order=2, mode='constant', cval=0)
+                # new_coord = coord + off_center
+                # # Disallow any whole-pixels moves on future iterations.
+                # allow_moves = False
```

Comment:
I'll look at it and refactor. Also the `shift_thresh` and `break_thresh` will be the same thing: if the code does not shift, it will break. Calculating the CoM without shifting makes no sense because the outcome will not change.